### PR TITLE
Fix KHI pgen. Add turbulence pgen doc.

### PR DESCRIPTION
**Pull request by pgrete**
The KHI pgen for iprob=5 had a typo (in the energy density init and potentially also in the v_y component) with regard to the method paper and the implementation in Athena++ itself.
Results match now what's in the paper.

Also, the KHI pgen is now executed on device.

Bonus: More doc!

### CODE REVIEW

File: docs/turbulence.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+<modes>
+k_1_0	= +2
+k_1_1	= -1
+k_1_2	= +0
+k_2_0	= +1
+...
+```
+
+Following parameters can be changed to control both the initial state
```
Comment (bwoshea) [inline docs/turbulence.md:50]: The following parameters can be changed to control both the initial state:

File: docs/turbulence.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+...
+```
+
+Following parameters can be changed to control both the initial state 
+
+- `rho0` initial mean density
+- `p0` initial mean thermal pressure
+- `b0` initial mean magnetic field strength
+- `b_config`
+  - `0`: net flux case (uniform B_x)
```
Comment (bwoshea) [inline docs/turbulence.md:56]: isn't it uniform B_z?  (same comments for 1 and 2)

File: docs/turbulence.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+- `kpeak` peak wavenumber of the forcing spectrum. Make sure to update the wavemodes to match `kpeak`, see below.
+- `corr_time` autocorrelation time of the acceleration field (in code units).
+Using delta-in-time forcing, i.e., a very low value, is discouraged, see [Grete et al. 2018 ApJL](https://iopscience.iop.org/article/10.3847/2041-8213/aac0f5).
+- `rseed` random seed for the OU process. Only change for new simulation, but keep unchanged for restarting simulations.
+- `sol_weight` solenoidal weight of the acceleration field. `1.0` is purely solenoidal/rotational and `0.0` is purely dilatational/compressive. Any value between `0.0` and `1.0` is possible. The parameter is related to the resulting rotational power in the 3D acceleration field as
+`1. - ((1-sol_weight)^2/(1-2*sol_weight+3*sol_weight^2))`, see eq (9) in [Federrath et al. 2010 A&A](
+https://doi.org/10.1051/0004-6361/200912437).
+- `accel_rms` root mean square value of the acceleration (controls the "strength" of the forcing field)
+- `num_modes` number of wavemodes that are specified in the `<modes>` section of the parameter file.
```
Comment (bwoshea) [inline docs/turbulence.md:76]: I think it's useful to very briefly explain the point of the <modes> section of the parameter file.  It's relatively easy to understand if you look at the source code, but documenting it saves time!

File: docs/turbulence.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+In order to generate a full set of modes run the `inputs/generate_fmturb_modes.py` script and replace
+the corresponding parts of the parameter file with the output of the script.
+Within the script, the top three variables (`k_peak`, `k_high`, and `k_low`) need to be adjusted in
```
Comment (bwoshea) [inline docs/turbulence.md:79]: maybe mention that k_peak in the script should be the same as kpeak in the parameter file.

File: src/pgen/kh.cpp [hunk 5]
```diff
@@ -32,61 +32,32 @@
...
-          }
-          // Pressure scaled to give a sound speed of 1 with gamma=1.4
-          u(IEN, k, j, i) =
-              2.5 / gm1 +
-              0.5 * (SQR(u(IM1, k, j, i)) + SQR(u(IM2, k, j, i))) / u(IDN, k, j, i);
-        }
-      }
-    }
-  }
+  //--- iprob=1. This was the classic, unresolved K-H test.
```
Comment (bwoshea) [inline src/pgen/kh.cpp:60]: is it worth throwing a special error for this?  i.e., "this version of the test is unresolved; try iprob >= 2"

**Review by bwoshea — APPROVED** [review]
Overall looks good, I have very minor comments.

File: docs/turbulence.md (hunk 2 above)
Comment (pgrete) [reply docs/turbulence.md:56]: Just double checked. It is B_x, see https://github.com/parthenon-hpc-lab/athenapk/blob/7ea10bbd55c7353acf5d2f6350ddc1f3c9f515f1/src/pgen/turbulence.cpp#L273

File: docs/turbulence.md (hunk 4 above)
Comment (pgrete) [reply docs/turbulence.md:79]: Good point. done.

File: src/pgen/kh.cpp (hunk 5 above)
Comment (pgrete) [reply src/pgen/kh.cpp:60]: It's already caught at the bottom of the if statements, so any non-implemented `iprob` will raise an explicit error.

**Pull request merged**